packages/shared-models/src/ymodels.ts

```diff
@@ -505,6 +505,10 @@ export class YBaseCell<Metadata extends models.ISharedBaseCellMetadata>
     this.ymodel = ymodel;
     const ysource = ymodel.get('source');
     this._prevSourceLength = ysource ? ysource.length : 0;
+
+    this.ystate = new Y.Map();
```

Comment:
What is the difference between `ystate` and `ymodel`?

Reply:
`ystate` would be extra state here, state that is not part of the actual json model. Though `ystate` is still an element of the `ymodel` in the implementation, so it's debatable.